fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItemSettings.java

```diff
@@ -39,7 +39,7 @@
 	 * @return this builder
 	 */
 	public FabricItemSettings equipmentSlot(EquipmentSlotProvider equipmentSlotProvider) {
-		FabricItemInternals.computeExtraData(this).equipmentSlot(equipmentSlotProvider);
+		((CustomItemSettingImpl<EquipmentSlotProvider>) CustomItemSettingImpl.EQUIPMENT_SLOT_PROVIDER).set(this, equipmentSlotProvider);
 		return this;
```

Comment:
```suggestion
		return customSetting(CustomItemSettingImpl.EQUIPMENT_SLOT_PROVIDER, equipmentSlotProvider);
```
I believe this would work too?